qt/src/SourceManager.cc

```diff
@@ -25,6 +25,7 @@
 #include <QFile>
 #include <QFileInfo>
 #include <QImageReader>
+#include <QListWidgetItem>
```

Comment:
I'm not really convinced it is worth the trouble to not rely on implicitly included headers, especially for situations like this were the implicit include is very unlikely to ever break. It's a lost battle to really consistently specify all necessary includes explicitly , and it just adds tons of include statements. As a maintainer of 150+ packages for Fedora, I very rarely get bitten by compilation failures because of internal header changes, and when I do, it is no big deal really. In my opinion the benefits of keeping the source files smaller outweigh the often theoretical problem of potential compilation failures in the future.

Reply:
But we can get compilation erros in future during upgrading to a next Qt version. 
I understand your point of view, but i think that we should include what we use. If compilation time is critical for even in .cpp files (i agree with your police about includes in headers), i can delete this include.

But i recommend left it.

Reply:
Yeah well but when upgrading to the next Qt version, this is likely the least of issues. (I'd rather hope that by then C++ modules come along and solve the include mess for good.) But in all the projects I've been or am active in, I've never seen this enforced, simply because it is very hard to really enforce. It is not about compilation time, the overhead of the preprocessor discarding the already included header is probably not that big, but simply about practicality - and about avoiding 100 lines of includes in each file.

Reply:
So... Must i remove it or not? :-)

Reply:
My preference is for minimizing the number of lines in the source files, so I'd remove it yes.

Reply:
Fixed.